src/main/java/com/amazon/opendistroforelasticsearch/sql/domain/Order.java

```diff
@@ -24,11 +24,17 @@
 	private String nestedPath;
 	private String name;
 	private String type;
+	private boolean isScriptField;
 
-	public Order(String nestedPath, String name, String type) {
+	public boolean isScriptField() {
+		return isScriptField;
+	}
+
+	public Order(String nestedPath, String name, String type, boolean isScriptField) {
```

Comment:
Several observations here:
1. for formal part, `Order` is not a field, so this flag does not belong in this class, or it should be called something else.
1. you use this information only in `DefaultQueryAction` to actually add the sorters. But there are more than 2 types of sorters, e.g. see `GeoDistanceSortBuilder`, so if in the future we need to add more, following this approach we will need to add another flag to this class, so at least this flag should be replaced by some enum type.
1. considering the above two, it seems to me this extra information is just a property of the field, and since you need it anyway, I think it belongs more on the `Field` object, and then ideally `Order` has a reference to that `Field` object, not just the name of it. I think what I suggest is a bigger change, but to me it seems the better alternative in terms of maintainability.

Reply:
> Several observations here:
> 
> 1. for formal part, `Order` is not a field, so this flag does not belong in this class, or it should be called something else.
> 2. you use this information only in `DefaultQueryAction` to actually add the sorters. But there are more than 2 types of sorters, e.g. see `GeoDistanceSortBuilder`, so if in the future we need to add more, following this approach we will need to add another flag to this class, so at least this flag should be replaced by some enum type.
> 3. considering the above two, it seems to me this extra information is just a property of the field, and since you need it anyway, I think it belongs more on the `Field` object, and then ideally `Order` has a reference to that `Field` object, not just the name of it. I think what I suggest is a bigger change, but to me it seems the better alternative in terms of maintainability.

For #2, sounds like we better replace this "tagged" class by subtyping.

Reply:
While I agree with that in general, I feel that this entire part should be rearchitected, so as of now to invest here might be a wasted effort.

Example: the problem that currently marked as ignored on integration test. When such query is converted into elastic's DSL, the entire Order by is dropped, because it should have order by in aggregation.

Reply:
I also agree with you, but I think the more special cases we add now the harder it will be to refactor later. I quickly checked the code, the `Order` constructor and the `addOrderBy` methods are used only in one place, so you're already modifying all the code paths that will be affected if you move the `isScript` logic to the `Field` class. Anyway, since you're working on a single bug fix, this is just a suggestion, not a blocker.